### DISCUSSION TITLE
Feature Request: Searchable Select Filters

### DISCUSSION THREAD

**Discussion by nastoychev** [laravel/nova-issues#6448] [Ideas & Feature Requests] [answered]
There is no built-in support for creating searchable filter components. Having searchable  filters would greatly enhance the usability and efficiency of the application, especially when dealing with large datasets.

We propose the introduction of a searchable filter logic which we can apply like:

```php
    public function filters(NovaRequest $request): array {
        return [
            SomeCustomSelectFilter::make()->searchable()
        ];
    }
```

It should look at the end similar to SelectField::make(...)->filterable()->searchable() and I guess you can reuse this Vue component.

Please note, we prefer searchable components that display all options, not just a single input field, which start showing options as we type. Thank you!

✅ The right searchable way! 🥇 
![Screenshot 2024-06-26 171250](https://github.com/laravel/nova-issues/assets/41673191/b10e0fbb-3ed8-4e68-8313-33514229ca72)

**Comment by jeremynikolic** [comment, marked as answer]
Thanks for suggesting this feature  @nastoychev, we did look into it and you can expect it to come as part of the next major release 👍

**Comment by crynobone** [comment]
Released with Laravel Nova 5.0.0

Feel free to open up a new issue if you're still experiencing this problem on the latest version.

**Reply by nastoychev** [reply]
This is looking great so far! However, we’ve received feedback from our departments that they would really appreciate having all the options displayed in the filter from the start. It would greatly enhance the user experience if it worked similarly to how **SelectField::make(...)->filterable()->searchable()** behaves, where users can quickly see and search through all available options.

I think a potential future issue could be handling a large set of options. To address this, here’s an example of how we can implement virtual scrolling to manage performance effectively: [virtual-scroll.zip](https://github.com/user-attachments/files/18572457/virtual-scroll.zip)

Could you kindly consider this and share if there are more options or solutions available? Thank you for your work on this!
